_posts/2019-03-06-release-1.8.0.md

```diff
@@ -0,0 +1,132 @@
+for more details.
+
+
+## New Features and Improvements
+
+* **New Support for Schema Migration when restoring Savepoints**: With Flink
+  1.7.0 we added support for changing the schema of state when using the
+  `AvroSerializer`
+  ([FLINK-10605](https://issues.apache.org/jira/browse/FLINK-10605)). With
+  Flink 1.8.0 we made great progress migrating all built-in `TypeSerializers`
+  to a new serializer snapshot abstraction that theoretically allows schema
+  migration. Of the serializers that come with Flink, we now support schema
+  migration for the `PojoSerializer`
+  ([FLINK-11485](https://issues.apache.org/jira/browse/FLINK-11485)) and Java
+  `EnumSerializer`
+  ([FLINK-11334](https://issues.apache.org/jira/browse/FLINK-11334)), as well
+  as for Kryo in limited cases
+  ([FLINK-11323](https://issues.apache.org/jira/browse/FLINK-11323)).
+
+
+* **Continuous cleanup of old state based on TTL
+  ([FLINK-7811](https://issues.apache.org/jira/browse/FLINK-7811))**: We
+  introduced TTL (time-to-live) for Keyed state in Flink 1.6
+  ([FLINK-9510](https://issues.apache.org/jira/browse/FLINK-9510)).  This
+  feature allowed to clean up and make inaccessible keyed state entries when
+  accessing them. In addition state would now also being cleaned up when
+  writing a savepoint/checkpoint.
+
+  Flink 1.8 introduces continous cleanup of old entries for both the RocksDB
+  state backend
+  ([FLINK-10471](https://issues.apache.org/jira/browse/FLINK-10471)) and the heap
+  state backend
+  ([FLINK-10473](https://issues.apache.org/jira/browse/FLINK-10473)). This means
+  that old entries (according to the ttl setting) are continously being cleanup
+  up.
+
+* **New KafkaDeserializationSchema that give direct access to ConsumerRecord
+  ([FLINK-8354](https://issues.apache.org/jira/browse/FLINK-8354))**: For the
+  Flink `KafkaConsumers`, we introduced a new `KafkaDeserializationSchema` that
+  gives direct access to the Kafka `ConsumerRecord`. This now allows access to
+  all data that Kafka provides for a record. This
+  subsumes`KeyedSerializationSchema` functionality, which is deprecated but
+  still available for now.
+
+## Important Changes
+
+* **Changes to bundling of Hadoop libraries with Flink
+  ([FLINK-11266](https://issues.apache.org/jira/browse/FLINK-11266))**:
+  Convenience binaries that include hadoop are no longer released.
+
+  If a deployment relies on `flink-shaded-hadoop2` being included in
+  `flink-dist`, then it must be manually downloaded and copied into the /lib
+  directory.  Alternatively, a Flink distribution that includes hadoop can be
+  built by packaging `flink-dist` and activating the `include-hadoop` maven
+  profile.
+
+  As hadoop is no longer included in `flink-dist` by default, specifying
+  `-DwithoutHadoop` when packaging `flink-dist` no longer impacts the build.
+
+* **FlinkKafkaConsumer will now filter restored partitions based on topic
```

Comment:
This one is a bit hard to digest if you are not familiar with the context of what is in the ticket. Maybe briefly explain based on an example of a topic rename what the new expected behavior would be in contrast with the old one?

Reply:
Would this additional paragraph help?
```
Consider this example: if you had a Kafka Consumer that was consuming
from topic `A`, you did a savepoint, then changed your Kafka consumer
to instead consume from topic `B`, and then restarted your job from
the savepoint. Before this change, your consumer would now consume
from both topic `A` and `B` because it was stored in state that the
consumer was consuming from topic `A`. With the change, your consumer
would only consume from topic `B` after restore because we filter the
topics that are stored in state using the configured topics.
```